src/main.rs

```diff
@@ -106,41 +154,55 @@ fn write_state(infile: &str, outfile: &str) {
         min_closeness,
         max_closeness,
     };
+
+    // Save all changes done to the cache
+    // TODO(asmie): better error handling - after refactorization of this function
+    geo_cache.save().await.expect("could not save geoip cache");
+
     let joutput = serde_json::to_string(&state).unwrap();
     fs::write(outfile, joutput).unwrap();
 }
 
-fn main() {
+#[tokio::main]
+async fn main() {
     let args: Vec<String> = env::args().collect();
-    if args.len() != 3 {
-        println!("\n\nUsage is: cargo run <in-sample.json> <out-state.json> ");
+
+    // TODO(asmie): could be refactored to use eg. clap crate.
```

Comment:
okay
just a thought - we could have a default geoip-cache.json location defined
throw that default file in `~/ziggurat/utils/crunchy/geoip-cache.json` e.g.

Reply:
Will be in config file in next PRs.